### ISSUE TITLE
L_diff

### ISSUE THREAD

**Issue by JunSun11** [hectornieto/pyTSEB#56]
Hello, your code is exquisite. But I encountered an issue while running the test data for TSEB-2T (as follows). I hope you can help me solve it. Thank you very much
```
Iteration 0, max. L diff: inf
Iteration 1, max. L diff: inf
D:\Desktop\pyTSEB-master\pyTSEB\TSEB.py:1956: RuntimeWarning: invalid value encountered in divide
  L_diff = np.asarray(np.fabs(L - L_old) / np.fabs(L_old), dtype=np.float32)
Iteration 2, max. L diff: inf
Iteration 3, max. L diff: inf
Iteration 4, max. L diff: inf
Iteration 5, max. L diff: inf
Traceback (most recent call last):
  File "D:\Desktop\pyTSEB-master\TSEB_local_image_main.py", line 36, in <module>
    run_TSEB_from_config_file(config_file)
  File "D:\Desktop\pyTSEB-master\TSEB_local_image_main.py", line 28, in run_TSEB_from_config_file
    setup.run(is_image=True)
  File "D:\Desktop\pyTSEB-master\pyTSEB\TSEBConfigFileInterface.py", line 274, in run
    model.process_local_image()
  File "D:\Desktop\pyTSEB-master\pyTSEB\PyTSEB.py", line 260, in process_local_image
    out_data = self.run(in_data, mask)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Desktop\pyTSEB-master\pyTSEB\PyTSEB.py", line 624, in run
    self._call_flux_model_veg(in_data, out_data, model_params, i)
  File "D:\Desktop\pyTSEB-master\pyTSEB\PyTSEB.py", line 1422, in _call_flux_model_veg
    out_data['u_friction'][i], out_data['L'][i], out_data['n_iterations'][i]] = TSEB.TSEB_2T(
                                                                                ^^^^^^^^^^^^^
  File "D:\Desktop\pyTSEB-master\pyTSEB\TSEB.py", line 463, in TSEB_2T
    i, l_queue, l_converged, l_diff_max = monin_obukhov_convergence(L,
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Desktop\pyTSEB-master\pyTSEB\TSEB.py", line 3761, in monin_obukhov_convergence
    l_diff_max = np.max(_L_diff(l_queue[0][i], l_queue[1][i]))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<__array_function__ internals>", line 200, in amax
  File "E:\python\Lib\site-packages\numpy\core\fromnumeric.py", line 2820, in amax
    return _wrapreduction(a, np.maximum, 'max', axis, None, out,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\python\Lib\site-packages\numpy\core\fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
zero-size array to reduction operation maximum which has no identity
```

**Comment by hectornieto** [comment]
Dear JunSun, thanks for your comments.

Please provide more information about the inputs and configuration file you are using

**Comment by JunSun11** [comment]
Thank you for your reply to my comment. In this email, I have packaged my Input and Config_LocalImage.txt.


假面超人
***@***.***



&nbsp;




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "hectornieto/pyTSEB"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年5月25日(星期四) 晚上7:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hectornieto/pyTSEB] L_diff (Issue #56)





 
Dear JunSun, thanks for your comments.
 
Please provide more information about the inputs and configuration file you are using

**Comment by hectornieto** [comment]
I could not find the attachments but I think I found the issue. 

Please try with the new code updated after this commit [f114761](https://github.com/hectornieto/pyTSEB/commit/f114761396875d4ec4c9b91fe711c52319ecfb31)

**Comment by JunSun11** [comment]
Thank you very much for your answer to my question. I am very sorry that I did not study this model again due to physical illness some time ago and did not reply to your email. I think you've solved that problem for me. Good luck to you.




假面超人
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "hectornieto/pyTSEB"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年7月6日(星期四) 晚上7:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [hectornieto/pyTSEB] L_diff (Issue #56)





 
Closed #56 as completed.
